server/workflow_template_server.go

```diff
@@ -0,0 +1,334 @@
+		Labels:   converter.APIKeyValueToLabel(req.WorkflowTemplate.Labels),
+	}
+	workflowTemplate, err = client.CreateWorkflowTemplate(req.Namespace, workflowTemplate)
+	if err != nil {
+		return nil, err
+	}
+	req.WorkflowTemplate.Uid = workflowTemplate.UID
+	req.WorkflowTemplate.Version = workflowTemplate.Version
+
+	return req.WorkflowTemplate, nil
+}
+
+func (s *WorkflowTemplateServer) CreateWorkflowTemplateVersion(ctx context.Context, req *api.CreateWorkflowTemplateRequest) (*api.WorkflowTemplate, error) {
+	client := ctx.Value("kubeClient").(*v1.Client)
+	allowed, err := auth.IsAuthorized(client, req.Namespace, "create", "argoproj.io", "workflows", req.WorkflowTemplate.Name)
+	if err != nil || !allowed {
+		return nil, err
+	}
+
+	workflowTemplate := &v1.WorkflowTemplate{
+		UID:      req.WorkflowTemplate.Uid,
+		Name:     req.WorkflowTemplate.Name,
+		Manifest: req.WorkflowTemplate.Manifest,
+		Labels:   converter.APIKeyValueToLabel(req.WorkflowTemplate.Labels),
+	}
+
+	workflowTemplate, err = client.CreateWorkflowTemplateVersion(req.Namespace, workflowTemplate)
+	if err != nil {
+		return nil, err
+	}
+	req.WorkflowTemplate.Uid = workflowTemplate.UID
+	req.WorkflowTemplate.Name = workflowTemplate.Name
+	req.WorkflowTemplate.Version = workflowTemplate.Version
+
+	return req.WorkflowTemplate, nil
+}
+
+func (s *WorkflowTemplateServer) UpdateWorkflowTemplateVersion(ctx context.Context, req *api.UpdateWorkflowTemplateVersionRequest) (*api.WorkflowTemplate, error) {
+	client := ctx.Value("kubeClient").(*v1.Client)
+	allowed, err := auth.IsAuthorized(client, req.Namespace, "update", "argoproj.io", "workflows", req.WorkflowTemplate.Name)
+	if err != nil || !allowed {
+		return nil, err
+	}
+
+	workflowTemplate := &v1.WorkflowTemplate{
+		UID:      req.WorkflowTemplate.Uid,
+		Name:     req.WorkflowTemplate.Name,
+		Manifest: req.WorkflowTemplate.Manifest,
+		Version:  req.WorkflowTemplate.Version,
+	}
+
+	req.WorkflowTemplate.Uid = workflowTemplate.UID
+	req.WorkflowTemplate.Name = workflowTemplate.Name
+	req.WorkflowTemplate.Version = workflowTemplate.Version
+
+	return req.WorkflowTemplate, nil
+}
+
+func (s *WorkflowTemplateServer) GetWorkflowTemplate(ctx context.Context, req *api.GetWorkflowTemplateRequest) (*api.WorkflowTemplate, error) {
+	client := ctx.Value("kubeClient").(*v1.Client)
+	allowed, err := auth.IsAuthorized(client, req.Namespace, "get", "argoproj.io", "workflows", "")
+	if err != nil || !allowed {
+		return nil, err
+	}
+
+	workflowTemplate, err := client.GetWorkflowTemplate(req.Namespace, req.Uid, req.Version)
+	if err != nil {
+		return nil, err
+	}
+
+	return apiWorkflowTemplate(workflowTemplate), nil
+}
+
+func (s *WorkflowTemplateServer) CloneWorkflowTemplate(ctx context.Context, req *api.CloneWorkflowTemplateRequest) (*api.WorkflowTemplate, error) {
+	client := ctx.Value("kubeClient").(*v1.Client)
+	allowed, err := auth.IsAuthorized(client, req.Namespace, "get", "argoproj.io", "workflows", "")
```

Comment:
I think only the `"create"` check is sufficient here since that's the highest permission.

Reply:
Fixed.